docs/backends/geoip.rst

```diff
@@ -142,95 +146,129 @@ that the ‐ before certain keys is part of the syntax.
-  records.
+:%cc:  ISO3166 country code.
+:%cn:  ISO3166 continent code.
+:%af:  v4 or v6.
+:%re:  Region code
+:%na:  AS organization name (spaces are converted to _)
+:%as:  AS number
+:%ci:  City name
+:%loc: LOC record style expansion of location
+:%lat: Decimal degree latitude
+:%lon: Decimal degree longitude
+
+These placeholders disable caching for the record completely:
+
+:%yy: Year
+:%mos: Month name
+:%mo: Month
+:%wds: Weekday name
+:%wd: Weekday
+:%dd: Year day
+:%hh: Hour
+:%ip: Client IP address
+:%ip4: Client IPv4 address
+:%ip6: Client IPv6 address
+
+.. versionadded:: 4.2.0
+
+  These placeholder have been added in version 4.2.0:
+
+  - %lat, %lon, %loc to expand for geographic location, if available in backend. %loc in particular can be safely used with LOC record type.
+  - %ip4 and %ip6 that will expand to the IP address when AFI matches, and empty otherwise. Can be particularly used with A and AAAA record types.
+
+.. versionadded:: 4.1.0
+
+  These placeholder have been added in version 4.1.0:
+
+  - %cc = 2 letter country code
+
+Using the ``weight`` attribute
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use record attributes to define weight.
+If this is given, only one record is chosen randomly based on the weight.
+
+Probability is calculated by summing up the weights and dividing each weight with the sum.
+
+Responses to queries
+~~~~~~~~~~~~~~~~~~~~
+
+If the record which a service points to exists under "records" then it is returned as a direct answer.
+If it does not exist under "records" then it is returned as a CNAME.
+
+You can mix service and static records to produce the sum of these records, including apex record.
+For instance, this configuration will send the correct response both A and SOA queries:
+
+.. code-block:: yaml
+
+  domains:
+  - domain: example.com
+  - ttl: 300
+  - records:
+    geo.example.com:
+      - soa: ns1.example.com hostmaster.example.com 2014090125 7200 3600 1209600 3600
+      - ns: ns1.example.com
+      - a: 192.0.2.1
+    swe.eu.example.com:
+      - a: 192.0.2.2
+  - services:
+    geo.example.com: ['%co.%cn.example.com']
+
+If your services match wildcard records in your zone file then these will be returned as CNAMEs.
+This will only be an issue if you are trying to use a service record at the apex of your domain where you need other record types to be present (such as NS and SOA records).
+Per :rfc:`2181`, CNAME records cannot appear in the same label as NS or SOA records.
+
+.. versionchanged:: 4.2.0
+
+  Before, a record expanded to an empty value it could cause SERVFAIL.
+  Since 4.2.0 such expansions for non-TXT record types are not included in response.
+
+Caching and the GeoIP Backend
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The :ref:`packet-cache` and :ref:`query-cache` will cache the records with EDNS Client Subnet information, when provided in the response.
+Use of certain placeholders (described above) can disable record caching for certail resource records.
```

Comment:
this should probably mention that narrowing scope when serving more-specifics can have sharp corners.

Reply:
Can you suggest some sentences on how to describe this?

Reply:
s/for certail resource/for certain resource/